<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EngineDialog</class>
 <widget class="QDialog" name="EngineDialog">
  <property name="windowModality">
   <enum>Qt::WindowModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>450</width>
    <height>200</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Engine properties</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_1" stretch="0,1,1">
     <item>
      <widget class="QLabel" name="nameLabel">
       <property name="text">
        <string>Name</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="nameLine">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="executableLabel">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Path to the engine's executable file.&lt;/p&gt;&lt;p&gt;If you're trying to add an Flatpak version of the Doom engine, you need to enter the path where the Flatpak installed the executable, usually something like &lt;span style=&quot; color:#606060;&quot;&gt;&amp;quot;/var/lib/flatpak/app/org.zdoom.GZDoom/current/active/files/bin/gzdoom&amp;quot;&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="whatsThis">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Path to the engine's executable file.&lt;/p&gt;&lt;p&gt;If you're trying to add an Flatpak version of the Doom engine, you need to enter the path where the Flatpak installed the executable, usually something like &lt;span style=&quot; color:#606060;&quot;&gt;&amp;quot;/var/lib/flatpak/app/org.zdoom.GZDoom/current/active/files/bin/gzdoom&amp;quot;&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Executable path &lt;img src=&quot;:/Help.12.png&quot;/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="executableLine"/>
     </item>
     <item>
      <widget class="QPushButton" name="selectExecutableBtn">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>24</height>
        </size>
       </property>
       <property name="text">
        <string>Browse</string>
       </property>
       <property name="icon">
        <iconset resource="../Resources/Resources.qrc">
         <normaloff>:/Browse.png</normaloff>:/Browse.png</iconset>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="configDirLabel">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Directory where the engine's &lt;span style=&quot; color:#686868;&quot;&gt;.ini&lt;/span&gt; or &lt;span style=&quot; color:#686868;&quot;&gt;.cfg&lt;/span&gt; files are.&lt;/p&gt;&lt;p&gt;On Windows this will be the same directory where the &lt;span style=&quot; color:#686868;&quot;&gt;.exe&lt;/span&gt; file is.&lt;br/&gt;The only exception is the latest GZDoom (thanks Graph!), which stores them in &lt;span style=&quot; color:#686868;&quot;&gt;&amp;quot;Documents\My Games\GZDoom&amp;quot;&lt;/span&gt;.&lt;br/&gt;On Linux this will probably be somewhere in your home, commonly &lt;span style=&quot; color:#686868;&quot;&gt;&amp;quot;~/.config/engine_name&amp;quot;&lt;/span&gt; or &lt;span style=&quot; color:#686868;&quot;&gt;&amp;quot;~/.local/share/engine_name&amp;quot;.&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; color:#ff0000;&quot;&gt;Red text&lt;/span&gt; means the directory doesn't exist and you should fix it.&lt;br/&gt;&lt;span style=&quot; color:#008000;&quot;&gt;Green text&lt;/span&gt; means the directory doesn't exist, but it will probably be created by the engine on the first start.&lt;/p&gt;&lt;p&gt;If you are not sure, leave it at the auto-detected value.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="whatsThis">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Directory where the engine's &lt;span style=&quot; color:#686868;&quot;&gt;.ini&lt;/span&gt; or &lt;span style=&quot; color:#686868;&quot;&gt;.cfg&lt;/span&gt; files are.&lt;/p&gt;&lt;p&gt;On Windows this will be the same directory where the &lt;span style=&quot; color:#686868;&quot;&gt;.exe&lt;/span&gt; file is.&lt;br/&gt;The only exception is the latest GZDoom (thanks Graph!), which stores them in &lt;span style=&quot; color:#686868;&quot;&gt;&amp;quot;Documents\My Games\GZDoom&amp;quot;&lt;/span&gt;.&lt;br/&gt;On Linux this will probably be somewhere in your home, commonly &lt;span style=&quot; color:#686868;&quot;&gt;&amp;quot;~/.config/engine_name&amp;quot;&lt;/span&gt; or &lt;span style=&quot; color:#686868;&quot;&gt;&amp;quot;~/.local/share/engine_name&amp;quot;.&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; color:#ff0000;&quot;&gt;Red text&lt;/span&gt; means the directory doesn't exist and you should fix it.&lt;br/&gt;&lt;span style=&quot; color:#008000;&quot;&gt;Green text&lt;/span&gt; means the directory doesn't exist, but it will probably be created by the engine on the first start.&lt;/p&gt;&lt;p&gt;If you are not sure, leave it at the auto-detected value.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Config directory &lt;img src=&quot;:/Help.12.png&quot;/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="configDirLine"/>
     </item>
     <item>
      <widget class="QPushButton" name="selectConfigDirBtn">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>24</height>
        </size>
       </property>
       <property name="text">
        <string>Browse</string>
       </property>
       <property name="icon">
        <iconset resource="../Resources/Resources.qrc">
         <normaloff>:/Browse.png</normaloff>:/Browse.png</iconset>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="dataDirLabel">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Directory where the engine stores its data files (save files, demo files, screenshots, ...).&lt;/p&gt;&lt;p&gt;In most cases this will be the same directory as the Config directory,&lt;br/&gt;with the exception of the latest GZDoom on Windows which stores them in &lt;span style=&quot; color:#686868;&quot;&gt;&amp;quot;Saved Games\GZDoom&amp;quot;&lt;/span&gt;.&lt;/p&gt;&lt;p&gt;NOTE: On Linux this is the parent directory of all the data files, not a specific subdirectory like &lt;span style=&quot; color:#686868;&quot;&gt;savefiles&lt;/span&gt;.&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; color:#ff0000;&quot;&gt;Red text&lt;/span&gt; means the directory doesn't exist and you should fix it.&lt;br/&gt;&lt;span style=&quot; color:#008000;&quot;&gt;Green text&lt;/span&gt; means the directory doesn't exist, but it will probably be created by the engine on the first start.&lt;/p&gt;&lt;p&gt;If you are not sure, leave it at the auto-detected value.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="whatsThis">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Directory where the engine stores its data files (save files, demo files, screenshots, ...).&lt;/p&gt;&lt;p&gt;In most cases this will be the same directory as the Config directory,&lt;br/&gt;with the exception of the latest GZDoom on Windows which stores them in &lt;span style=&quot; color:#686868;&quot;&gt;&amp;quot;Saved Games\GZDoom&amp;quot;&lt;/span&gt;.&lt;/p&gt;&lt;p&gt;NOTE: On Linux this is the parent directory of all the data files, not a specific subdirectory like &lt;span style=&quot; color:#686868;&quot;&gt;savefiles&lt;/span&gt;.&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; color:#ff0000;&quot;&gt;Red text&lt;/span&gt; means the directory doesn't exist and you should fix it.&lt;br/&gt;&lt;span style=&quot; color:#008000;&quot;&gt;Green text&lt;/span&gt; means the directory doesn't exist, but it will probably be created by the engine on the first start.&lt;/p&gt;&lt;p&gt;If you are not sure, leave it at the auto-detected value.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Data directory &lt;img src=&quot;:/Help.12.png&quot;/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="dataDirLine"/>
     </item>
     <item>
      <widget class="QPushButton" name="selectDataDirBtn">
       <property name="text">
        <string>Browse</string>
       </property>
       <property name="icon">
        <iconset resource="../Resources/Resources.qrc">
         <normaloff>:/Browse.png</normaloff>:/Browse.png</iconset>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QLabel" name="familyLabel">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Different engines use different command line parameters. This determines which parameter set will be used.&lt;/p&gt;&lt;p&gt;If you are not sure, leave it at the auto-detected value.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="whatsThis">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Different engines use different command line parameters. This determines which parameter set will be used.&lt;/p&gt;&lt;p&gt;If you are not sure, leave it at the auto-detected value.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Engine family &lt;img src=&quot;:/Help.12.png&quot;/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="familyCmbBox">
       <property name="minimumSize">
        <size>
         <width>120</width>
         <height>0</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="autoDetectBtn">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Attempt to automatically detect the paths and properties from the currently set Executable path.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="whatsThis">
        <string>Attempt to automatically detect the paths and properties from the currently set Executable path.</string>
       </property>
       <property name="text">
        <string>  Auto-detect  </string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../Resources/Resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>EngineDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>EngineDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
